### PR TITLE
Add XLM-V to Model Doc

### PR DESCRIPTION
**Pull request by stefan-it**
Hi,

as discussed in #21330 it would be good to have an extra entry for the new XLM-V model in the Model Doc.

This PR adds it with some additional information about the model and conducted experiments with it.

### CODE REVIEW

File: docs/source/en/model_doc/xlm-v.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+> Large multilingual language models typically rely on a single vocabulary shared across 100+ languages.
+> As these models have increased in parameter count and depth, vocabulary size has remained largely unchanged.
+> This vocabulary bottleneck limits the representational capabilities of multilingual models like XLM-R.
+> In this paper, we introduce a new approach for scaling to very large multilingual vocabularies by
+> de-emphasizing token sharing between languages with little lexical overlap and assigning vocabulary capacity
+> to achieve sufficient coverage for each individual language. Tokenizations using our vocabulary are typically
+> more semantically meaningful and shorter compared to XLM-R. Leveraging this improved vocabulary, we train XLM-V,
+> a multilingual language model with a one million token vocabulary. XLM-V outperforms XLM-R on every task we
+> tested on ranging from natural language inference (XNLI), question answering (MLQA, XQuAD, TyDiQA), and
+> named entity recognition (WikiAnn) to low-resource tasks (Americas NLI, MasakhaNER).
```
Comment (sgugger) [inline docs/source/en/model_doc/xlm-v.mdx:32]: Should be in italics.

File: docs/source/en/model_doc/xlm-v.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Tips:
+
+- XLM-V is compatible with the XLM-RoBERTa model architecture, only model weights from [`fairseq`](https://github.com/facebookresearch/fairseq)
+  library had to be converted.
+- The `XLMTokenizer` implementation is used to load the vocab and performs tokenization.
+
+This model was contributed by [stefan-it](https://huggingface.co/stefan-it), including detailed experiments with XLM-V on downstream tasks.
```
Comment (sgugger) [inline docs/source/en/model_doc/xlm-v.mdx:40]: Could you point to one canonical checkpoint on the Hub as well?

**Review by sgugger — COMMENTED** [review]
Thanks for adding this!

**Comment by stefan-it** [comment]
CI is failing, I'm going to read the Flan-T5 PR (https://github.com/huggingface/transformers/pull/19892) to see how it should be done!

**Comment by sgugger** [comment]
You just need to add the model type (same as what you picked for the page in the doc) and name in the configuration_auto file. The PR you mention also does it :-)

File: docs/source/en/model_doc/xlm-v.mdx (hunk 1 above)
Comment (stefan-it) [reply docs/source/en/model_doc/xlm-v.mdx]: Fixed :)

File: docs/source/en/model_doc/xlm-v.mdx (hunk 2 above)
Comment (stefan-it) [reply docs/source/en/model_doc/xlm-v.mdx:42]: Thanks, I added a reference to `facebook/xlm-v-base` :)

**Comment by sgugger** [comment]
Failures are unrelated to this PR so merging!

**Review by sgugger — APPROVED** [review]

**Pull request merged**